pkg/controller/machine/kubeconfig.go

```diff
@@ -66,13 +68,23 @@ func (r *Reconciler) createBootstrapKubeconfig(name string) (*clientcmdapi.Confi
 	}
 
 	outConfig := infoKubeconfig.DeepCopy()
+	clusterContextName := ""
+	for key := range infoKubeconfig.Clusters {
+		clusterContextName = key
```

Comment:
So `clusterContextName` will get the last key as it always will be overwritten inside the loop?

Reply:
Yes, its because we have some validation down the line that fails if this has more than one key. Lemme add a comment for that.